cmd/lakectl/cmd/dbt.go

```diff
@@ -0,0 +1,266 @@
+		copySchemaWithDbtTables(cmd.Context(), continueOnError, continueOnSchemaExists, clientType, schema, toSchema, branchName, projectRoot, dbfsLocation)
+
+		if skipViews {
+			fmt.Println("skipping views")
+		} else {
+			createViews(projectRoot, toSchema)
+		}
+
+		fmt.Printf("create-branch-schema done! \n your new schema is %s\n you could now edit your dbt profile and run dbt on branch %s\n", toSchema, branchName)
+	},
+}
+
+func validateGenerateSchemaMacro(projectRoot string) {
+	p := path.Join(projectRoot, macrosDirName, generateSchemaName)
+	file, err := os.Open(p)
+	defer func() {
+		if err = file.Close(); err != nil {
+			DieErr(err)
+		}
+	}()
+	if err != nil {
+		if os.IsNotExist(err) {
+			DieFmt("missing macro generate_schema_name.sql macro, run 'lakectl dbt generate-macro' or add manually")
+		}
+		DieErr(fmt.Errorf("failed to read generate schema macro: %w", err))
+	}
+	if !containsLakectlIdentifier(file) {
+		// TODO(Guys): add link to docs
+		DieFmt("generate_schema_name does not contain lakectl addition. handle lakefs support to generate_schema_name.sql or use skip-views flag")
+	}
+}
+
+// copySchemaWithDbtTables create a copy of fromSchema and copies all dbt models materialized as table or incremental
+func copySchemaWithDbtTables(ctx context.Context, continueOnError, continueOnSchemaExists bool, clientType, fromSchema, toSchema, branchName, projectRoot, dbfsLocation string) {
+	client, clientDeferFunc := getMetastoreClient(clientType, "")
+	defer clientDeferFunc()
+	if len(toSchema) == 0 {
+		toSchema = branchName
+	}
+
+	err := metastore.CopyDB(ctx, client, client, fromSchema, toSchema, branchName, dbfsLocation)
+	switch {
+	case err != nil && !errors.Is(err, mserrors.ErrSchemaExists):
+		DieErr(err)
+	case errors.Is(err, mserrors.ErrSchemaExists) && continueOnSchemaExists:
+		fmt.Printf("schema %s already exists, proceeding with existing schema\n", toSchema)
+	case errors.Is(err, mserrors.ErrSchemaExists):
+		DieFmt("schema %s exists, change schema or use --continue-on-schema-exists flag\n", toSchema)
+	default:
+		fmt.Printf("created schema %s\n", toSchema)
+	}
+
+	if err := CopyDbtTables(ctx, client, continueOnError, projectRoot, fromSchema, toSchema, branchName, dbfsLocation); err != nil {
+		DieErr(err)
+	}
+}
+
+func createViews(projectRoot, toSchema string) {
+	os.Setenv(lakectlIdentifier, toSchema)
+	defer os.Unsetenv(lakectlIdentifier)
+	dbtCmd := exec.Command("dbt", "run", "--select", "config.materialized:view")
+	dbtCmd.Dir = projectRoot
+	if err := dbtCmd.Run(); err != nil {
+		DieFmt("failed creating views with err: %v\n", err)
+	}
+	fmt.Println("views created")
+}
+
+type model struct {
+	Schema string `json:"schema"`
+	Alias  string `json:"alias"`
+}
+
+// CopyDbtTables copies dbt tables (models materialized as tables or incremental) to given schema
+func CopyDbtTables(ctx context.Context, client metastore.Client, continueOnError bool, projectRoot, fromSchema, toSchema, branchName, dbfsLocation string) error {
```

Comment:
alternative approach was to write a function that just extract the information.
Have a function calling each step will also enable a quick way to test the code.
1. execute dbt -> json
2. parse json -> structure model / what we need to copy
3. copy tables

Reply:
Broke into two functions, one for the first two steps one for copy tables.